### PR TITLE
Fixes Red Card / Eject Pack interaction with Emergency Exit 

### LINKED ISSUE #5550: Emergency Exit + Eject Pack timing bug
### Description

When a Pokemon uses up it's own Eject Pack (say Typhlosion uses Overheat) on a Pokemon with Emergency Exit (Golisopod), Emergency Exit should activate and cancel out Eject Pack. Currently in the Expansion, both Pokemon switch out.
Showdown proof: https://replay.pokemonshowdown.com/gen9customgame-2227013955 (Typhlosion holds Eject Pack, you can even see it getting consumed, but still cancelled)
Showcasing how it works with Double Battles: 
https://replay.pokemonshowdown.com/gen9doublescustomgame-2227104763 - Typhlosion hits Golisopod into Emergency Exit range and it's Eject Pack still gets cancelled
https://replay.pokemonshowdown.com/gen9doublescustomgame-2227015758 - Typhlosion misses out on activating Golisopods Emergency Exit, thus activating its Eject Pack successfully. Ninjask then gets 1 more hit off, activating Golisopods Emergency Exit

### Version

1.9.3 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

._.terra._.

### LINKED ISSUE #5551: Emergency Exit + Red Card timing bug
### Description

If a Pokemon having Emergency Exit and holding a Red Card is attacked, activating it's Red Card, the opposing Pokemon will be switched out and Emergency Exit get's cancelled. See [in Bulbapedia](https://bulbapedia.bulbagarden.net/wiki/Red_Card).

Currently in the Expansion, both Pokemon switch out.

### Version

1.9.3 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

._.terra._.

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #5551
Fixes #5550

One test is removed because the interaction was wrong

### CODE REVIEW

File: include/battle_scripts.h [hunk 1]
```diff
@@ -418,6 +418,7 @@ extern const u8 BattleScript_EjectButtonActivates[];
 extern const u8 BattleScript_EjectPackActivate_Ret[];
 extern const u8 BattleScript_EjectPackActivate_End2[];
 extern const u8 BattleScript_EjectPackActivates[];
+extern const u8 BattleScript_EjectPackMissisTiming[];
```
Comment (Pawkkie) [inline include/battle_scripts.h:421]:
```suggestion
extern const u8 BattleScript_EjectPackMissesTiming[];
```

File: data/battle_scripts_1.s [hunk 2]
```diff
@@ -9628,6 +9628,13 @@ BattleScript_EjectPackActivates::
 	jumpifcantswitch BS_SCRIPTING, BattleScript_EjectButtonEnd
 	goto BattleScript_EjectPackActivate_Ret
 
+BattleScript_EjectPackMissisTiming::
```
Comment (Pawkkie) [inline data/battle_scripts_1.s:9631]:
```suggestion
BattleScript_EjectPackMissesTiming::
```

**Review by Pawkkie — CHANGES_REQUESTED** [review]
Just typo of missis vs misses, looks good otherwise!

**Comment by AlexOn1ine** [comment]
changes commited

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -6178,13 +6178,21 @@ static void Cmd_moveend(void)
...
+                                {
+                                    gBattlescriptCurrInstr = BattleScript_EjectPackActivates;
+                                    // Are these 2 lines below needed?
+                                    gProtectStructs[battler].statFell = FALSE;
+                                    gSpecialStatuses[gBattlerAttacker].preventLifeOrbDamage = TRUE;
+                                }
                             }
                             break; // Only the fastest Eject item activates
-                        }
+                            }
```
Comment (AlexOn1ine) [inline src/battle_script_commands.c:6195]:
```suggestion
                        }
```

**Pull request merged**